tests/furusiya/model/components/test_fighter.py

```diff
@@ -0,0 +1,65 @@
+from unittest.mock import Mock
+
+import pytest
+
+from model.components.fighter import Fighter
+from model.systems.fighter_system import FighterSystem
+
+
+@pytest.fixture
+def fighter():
+    yield Fighter(Mock(), 30, 5, 5, death_function=Mock())
+
+
+@pytest.fixture
+def bushslime():
+    yield Mock()
+
+
+@pytest.fixture
+def bushslime_fighter(bushslime):
+    bush_fighter = Fighter(bushslime, 15, 2, 2, death_function=Mock())
+    FighterSystem.set_fighter(bushslime, bush_fighter)
+    yield bush_fighter
+
+
+def test_take_damage(fighter):
+    old_hp = fighter.hp
+    fighter.take_damage(5)
+    assert fighter.hp == old_hp - 5
+
+
+def test_take_damage_ignore_negative(fighter):
+    old_hp = fighter.hp
+    fighter.take_damage(-5)
+    assert fighter.hp == old_hp
+
+
+def test_attack(fighter, bushslime, bushslime_fighter):
+    old_hp = bushslime_fighter.hp
+    fighter.attack(bushslime)
+    assert bushslime_fighter.hp < old_hp
+
+
+def test_attack_weapon(fighter, bushslime, bushslime_fighter):
+    fighter.weapon = Mock()
+    test_attack(fighter, bushslime, bushslime_fighter)
+    fighter.weapon.attack.assert_called_with(bushslime)
+
+
+def test_heal(fighter):
+    test_take_damage(fighter)
+    old_hp = fighter.hp
+    fighter.heal(3)
```

Comment:
Instead of magic numbers, I would use variables with clear names, like `expected_health_gain`